internal/expr/prepare.go

```diff
@@ -133,34 +142,47 @@ func prepareOutput(ti typeNameToInfo, p *outputPart) ([]fullName, []field, error
 	// Case 1: Star target cases e.g. "...&P.*".
 	if p.target[0].name == "*" {
 		info, _ := ti[p.target[0].prefix]
-
 		// Case 1.1: Single star i.e. "t.* AS &P.*" or "&P.*"
 		if len(p.source) == 0 || p.source[0].name == "*" {
-			pref := ""
-
-			// Prepend table name. E.g. "t" in "t.* AS &P.*".
-			if len(p.source) > 0 {
-				pref = p.source[0].prefix
-			}
+			switch info := info.(type) {
+			case *mapInfo:
+				return nil, nil, fmt.Errorf(`&%s.* cannot be used with a map output when no column names are specified or column name is *`, info.typ().Name())
```

Comment:
Changed.
By reading all columns do you mean put in a wildcard e.g. `SELECT * FROM ...`? I would be hesitant to do this considering that its generally bad practice to use wildcards, and if we removed this error and selected everything instead then there's a chance the user is doing by mistake (since it would be perfectly valid to do `&P.*` if `P` was a struct)